src/VisualStudio/CSharp/Impl/LanguageService/CSharpLanguageService_ICSharpProjectHost.cs

```diff
@@ -16,6 +17,15 @@ internal partial class CSharpLanguageService : ICSharpProjectHost
         public void BindToProject(ICSharpProjectRoot projectRoot, IVsHierarchy hierarchy)
         {
             var projectName = Path.GetFileName(projectRoot.GetFullProjectName()); // GetFullProjectName returns the path to the project file w/o the extension?
+            var projectFilename = AbstractLegacyProject.GetProjectFilePath(hierarchy);
+            var projectId = Workspace.ProjectTracker.GetOrCreateProjectIdForPath(projectFilename, projectName);
+
+            if (Workspace.Services.GetService<IDeferredProjectWorkspaceService>()?.IsDeferredProjectLoadEnabled ?? false)
+            {
+                var existingProject = Workspace.ProjectTracker.GetProject(projectId);
```

Comment:
We're doing it to close the project we may have created while the load was deferred, so that we only have one project for that id alive.  We favor the "real" project if it ever gets created.

Will comment if not accepted today.